huanyu-li/week7/tues/js/main.js

```diff
@@ -0,0 +1,155 @@
+
+const apiKey = 'ck6JfRaH5OyKQUHFxZ3BpB6bmqyGt9jlHHYqb4ez'
+
+const httpByNumber = function (count) {
+    return `https://api.nasa.gov/planetary/apod?api_key=${apiKey}&count=${count}`
+}
+
+const httpByDate = function (date) {
+    return `https://api.nasa.gov/planetary/apod?api_key=${apiKey}&date=${date}`
+}
+
+const httpByTwoDates = function (start, end) {
+    return `https://api.nasa.gov/planetary/apod?api_key=${apiKey}&start_date=${start}&end_date=${end}`
+}
+
+
+
+$(function () {
+    // DOM ready
+    $('#btn').on('click', function () {
+        $('#details').empty().hide()
+        $('#results').empty().show()
+        const query = $('#query').val()
+        getImagesResults(query)
+    }) // query button handler
+
+    $('#dateBtn').on('click', function () {
+        $('#details').empty().hide()
+        $('#results').empty().show()
+        const startDate = $('#startDate').val()
+        const endDate = $('#endDate').val()
+        getImagesByDate(startDate, endDate)
+    }) // dates button handler
+
+    $('#todayBtn').on('click', function () {
+        $('#results').empty().hide()
+        $('#details').empty().show()
+        const today = $('#today').val()
+        getDetails(today)
+    }) // dates button handler
+
+    $('#searchType').change(function(){
+        const type = $(this).val()
+        if(type==='0'){
+            $('#searchByNumber').show()
+            $('#searchByDateRange').hide()
+            $('#searchByDate').hide()
+        } else if (type==='1') {
+            $('#searchByNumber').hide()
+            $('#searchByDateRange').show()
+            $('#searchByDate').hide()
+        } else {
+            $('#searchByNumber').hide()
+            $('#searchByDateRange').hide()
+            $('#searchByDate').show()
+        }
+    })
+}) //DOM ready
+
+const getImagesResults = function (queryNumber) {
+    axios.get(httpByNumber(queryNumber))
+        .then(function (res) {
+            //success
+            const data = res.data
+            showImages(data)
+        })
+        .catch(function (err) {
+            // err
+            showError(err)
+        })
+} // getImagesResults
+
+const getDetails = function (queryDate) {
+    axios.get(httpByDate(queryDate))
+        .then(function (res) {
+         
+            const data = res.data
+            $('#results').hide()
+            $('#details').empty().append(`
+                <h3>${data.title}</h3>
+                <img class="back" src="${data.hdurl}" alt="${data.title}">
+                <em>By: ${data.copyright}, on: ${data.date}</em>
+                <br><br>
+                <strong>Explanation:</strong>
+                <p> ${data.explanation}</p>
+                <br>
+                <button class="back">Back</button>
+                `).show()
+
+            $('.back').on('click', function () {
+         
+                $('#details').hide()
+                $('#results').show()
+            }) // back button handler
+
```

Comment:
when i click on the image, it disappears because it has the same name as the back button. consider changing it to an ID or give it a different name